src/Integration/LocalServices/ISourceControlledFileSystem.cs

```diff
@@ -19,15 +19,18 @@
  */
 
 using System;
+using System.IO.Abstractions;
 
 namespace SonarLint.VisualStudio.Integration
 {
     /// <summary>
     /// Provides abstraction over the SCC file management. The files are queued up using <see cref="QueueFileWrite(string, Func{bool})"/>
     /// until <see cref="WriteQueuedFiles"/> which will call the SCC to check out the file and will run the queued up write operations.
     /// </summary>
-    internal interface ISourceControlledFileSystem : IFileSystem
+    internal interface ISourceControlledFileSystem : ILocalService
     {
+        IFileSystem FileSystem { get; }
```

Comment:
You've changed `ISourceControlledFileSystem` so it is no longer a `IFileSystem` (which is good). I think you should separate them completely i.e. remove this property. A quick look at the code suggests that there are only two product classes that would be affected by this (and it looks like one of them is only accessing the file system in assertions so it probably doesn't need an `IFileSystem` at all).

Let's treat `IFileSystem` as purely an abstraction for testing that can be injected into the constructor of classes that need it.